src/pool/disjoint/pool_disjoint.cpp

```diff
@@ -29,22 +29,15 @@ disjoint_pool_initialize(umf_memory_provider_handle_t provider,
     config.CurPoolSize = pub_params->CurPoolSize;
     config.PoolTrace = pub_params->PoolTrace;
 
-    struct disjoint_memory_pool *pool_data =
-        (struct disjoint_memory_pool *)malloc(
-            sizeof(struct disjoint_memory_pool));
-    if (!pool_data) {
-        fprintf(stderr, "cannot allocate memory for metadata\n");
-        abort();
-    }
-
+    struct disjoint_memory_pool *pool_data = new struct disjoint_memory_pool;
```

Comment:
You should also update the finalize function (free -> delete).

Reply:
Right, I have forgotten about that. Done:
https://github.com/oneapi-src/unified-memory-framework/pull/19
Thanks!